Simulator/game_round.py

```diff
@@ -195,6 +167,7 @@ def start_round(self):
         for player in self.PLAYERS.values():
             if player:
                 player.start_round(self.current_round)
+        self.decide_player_combat(self.PLAYERS)

```

Comment:
We probably want this before the start_round call because if we put this here, I think it won't update the player_vector with the correct information for the first action taken on the agent's turn.